GOVERNANCE.md

```diff
@@ -0,0 +1,110 @@
+Keycloak. We strongly believe if Keycloak would support all use cases by default it would become bloated and hard to use.
+
+Keycloak aims to be opinionated and make it as easy as possible to achieve the common use cases, while still
+enabling the less common use cases through custom extensions.
+
+
+## Projects
+
+Keycloak consists of several projects:
+
+* [Keycloak](https://github.com/keycloak/keycloak) - Keycloak Server and Java adapters
+* [Keycloak Documentation](https://github.com/keycloak/keycloak-documentation) - Documentation for Keycloak
+* [Keycloak QuickStarts](https://github.com/keycloak/keycloak-quickstarts) - QuickStarts for getting started with Keycloak
+* [Keycloak Containers](https://github.com/keycloak/keycloak-containers) - Container images for Keycloak
+* [Keycloak Gatekeeper](https://github.com/keycloak/keycloak-gatekeeper) - Proxy service to secure apps and services with Keycloak
+* [Keycloak Node.js Connect](https://github.com/keycloak/keycloak-nodejs-connect) - Node.js adapter for Keycloak
+* [Keycloak Node.js Admin Client](https://github.com/keycloak/keycloak-nodejs-admin-client) - Node.js library for Keycloak Admin REST API
+
+The same governance model applies to all projects. However, the list of maintainers may vary per project. 
+
+
+## Maintainers
+
+The list of maintainers can be found in the [MAINTAINERS.md](MAINTAINERS.md) file in the repository for the individual 
+projects listed in the [Projects](#projects) section.
+
+### Maintainer Responsibilities
+
+A maintainer is someone who has shown deep knowledge of vision, features and codebase. It is their 
+responsibility to drive the project forward, encourage collaboration and contributions, and generally help the 
+community.
+
+Responsibilities of a maintainer include, but are not limited to:
+
+* Engage in design discussions
+* Actively monitor mailing lists, user forum and chat
+* Contribute high quality code
+* Maintain deep knowledge of vision, features and codebase
+* Review pull requests either personally or delegate to experts in the relevant area
+* Helping the community
+
+### Becoming a Maintainer
+
+To become a maintainer, you need to demonstrate the following:
+
+* Good understanding of vision, features and codebase
+* Contribution of larger features
+* Contribution of bug fixes
+* Participation in design discussions
+* Participation in pull request reviews
+* Ability to collaborate with the team
+* Helping the community
+
+A new maintainer must be proposed by sending an email to the [developer mailing list](https://lists.jboss.org/mailman/listinfo/keycloak-dev).
+The email should include evidence of the above list.
+
+The existing maintainers will then discuss the proposal. If anyone objects or wants more information, the maintainers 
+will reach out to the nominee directly for further discussion. 
+
+For the nominee to be accepted as a maintainer at least 2/3 of existing maintainers have to approve the nominee.
+
+
+### Changes in Maintainership
```

Comment:
@stianst thinking for a bit, maybe would be nice to some criteria to these changes? We know what's required to do to become a maintainer, but not the criteria to be removed.

So, for example, something like:
```
Maintainers can be removed if at least 2/3 of existing maintainers agree. Based on the following criteria:
* Lack of activity on after X months after a warning from one of the maintainers: PR reviews, design discussions or mailing list
* Lack of interactivity with the community
* Not following the code of conduct (if we have some)

The person requesting the removal of the maintainer should present evidence and arguments on why that person should be removed privately.
```
The items below are just examples and ideas to align the expectations around this.

Reply:
I'd rather just leave it at the discretion of the maintainers on what conditions a maintainer would be removed for. Could potentially add something along the lines of "If there are any concerns about a maintainer contact the project lead. The project lead will then discuss the matter with the maintainer. If any issues can not be resolved the maintainer would be removed if at least 2/3 of existing maintainers agree."

Reply:
Sure np, that's not a deal breaker. Marking it as resolved, because it looks good to me as is.